static/app/views/automations/components/dataConditionNodes.tsx

```diff
@@ -0,0 +1,86 @@
+import {createContext, useContext} from 'react';
+import styled from '@emotion/styled';
+
+import NumberField from 'sentry/components/forms/fields/numberField';
+import SelectField from 'sentry/components/forms/fields/selectField';
+import {t} from 'sentry/locale';
+import {
+  DataConditionType,
+  type NewDataCondition,
+} from 'sentry/types/workflowEngine/dataConditions';
+import AgeComparisonNode from 'sentry/views/automations/components/actionFilters/ageComparison';
+import IssueOccurrencesNode from 'sentry/views/automations/components/actionFilters/issueOccurrences';
+
+interface DataConditionNodeProps {
+  condition: NewDataCondition;
+  condition_id: string;
+  onUpdate: (condition: Record<string, any>) => void;
+}
+
+export const DataConditionNodeContext = createContext<DataConditionNodeProps | null>(
+  null
+);
+
+export function useDataConditionNodeContext(): DataConditionNodeProps {
+  const context = useContext(DataConditionNodeContext);
+  if (!context) {
+    throw new Error(
+      'useDataConditionNodeContext was called outside of DataConditionNode'
+    );
+  }
+  return context;
+}
+
+type DataConditionNode = {
+  label: string;
+  dataCondition?: React.ReactNode;
+};
+
+export const dataConditionNodesMap: Partial<
+  Record<DataConditionType, DataConditionNode>
+> = {
+  [DataConditionType.FIRST_SEEN_EVENT]: {
+    label: t('A new issue is created'),
+  },
+  [DataConditionType.REGRESSION_EVENT]: {
+    label: t('A resolved issue becomes unresolved'),
+  },
+  [DataConditionType.REAPPEARED_EVENT]: {
+    label: t('An issue escalates'),
+  },
+  [DataConditionType.AGE_COMPARISON]: {
+    label: t('Compare the age of an issue'),
+    dataCondition: <AgeComparisonNode />,
+  },
+  [DataConditionType.ISSUE_OCCURRENCES]: {
+    label: t('Issue frequency'),
+    dataCondition: <IssueOccurrencesNode />,
+  },
+};
```

Comment:
This could be an actual `Map()` if you want! Essentially defined as an array of tuples rather than an object. Doesn't change much in the end, up to you.

```js
export const dataConditionNodesMap = new Map<DataConditonType, DataConditionNode>([
  [DataConditionType.FIRST_SEEN_EVENT, { label: t('A new issue is created') }],
  // ...
])
```

Reply:
i didn't realize there was a difference between `Records` and `Maps` 😳  updated in https://github.com/getsentry/sentry/pull/89233/commits/c3b26dc656daeca63eb34f33b18ab0b48f40ce34